### PR TITLE
emacs: update to 29.2, add tree-sitter-langs

### PR DESCRIPTION
**Pull request by citrus-it**
Pulling in updates to emacs from omnios, including tree-sitter integration and compiled grammars.

### CODE REVIEW

File: build/emacs/build.sh [hunk 1]
```diff
@@ -12,18 +12,21 @@
...
 # }}}
 
-# Copyright 2023 OmniOS Community Edition (OmniOSce) Association.
+# Copyright 2024 OmniOS Community Edition (OmniOSce) Association.
 
 . ../../lib/build.sh
 
 PROG=emacs
-VER=29.1
+VER=29.2
```
Comment (zeeshanlakhani) [inline build/emacs/build.sh]: Minor q @citrus-it, should we move to 29.3 (that's what I see in Omni's package search ~ [ooce/editor/emacs@29.3-151050.0:20240504T140742Z](https://pkg.omnios.org/r151050/extra/info/0/ooce%2Feditor%2Femacs%4029.3%2C5.11-151050.0%3A20240504T140742Z)).

File: build/emacs/build.sh [hunk 2]
```diff
@@ -49,6 +52,7 @@ CONFIGURE_OPTS="
     --prefix=$PREFIX
     --without-x
     --with-gif=no
+    --with-tree-sitter
```
Comment (zeeshanlakhani) [inline build/emacs/build.sh:55]: being that we already have jansson in the /opt/ooce/lib/amd (64-bit), we can also no `--with-json` for native support.

File: build/tree-sitter-langs/patches/illumos.patch [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     ('windows-nt "windows")
++    ('usg-unix-v "illumos")
+     (_ (error "Unsupported system-type %s" system-type))))
+ 
+ (defconst tree-sitter-langs--suffixes '(".dylib" ".dll" ".so")
+@@ -410,6 +411,26 @@ from the current state of the grammar repo, without cleanup."
+                "-I" "src"
+                "src/parser.c"
+                "-o" (format "%sbin/%s.so" tree-sitter-langs-grammar-dir lang-symbol)))))
++           ((memq system-type '(usg-unix-v))
```
Comment (zeeshanlakhani) [inline build/tree-sitter-langs/patches/illumos.patch:26]: \m/

File: build/emacs/build.sh (hunk 1 above)
Comment (citrus-it) [reply build/emacs/build.sh]: Yes, I missed that commit somehow.

File: build/emacs/build.sh (hunk 2 above)
Comment (citrus-it) [reply build/emacs/build.sh:55]: Done.

**Review by zeeshanlakhani — APPROVED** [review]

**Pull request merged**